src/functions/authoriser/application/__tests__/transformMethodArn.spec.ts

```diff
@@ -0,0 +1,66 @@
+import * as transformMethodArn from '../transformMethodArn';
+
+describe('transformMethodArn.toAllVerbsAndAllResources', () => {
+  const sut = transformMethodArn.toAllVerbsAndAllResources;
+
+  it('changes httpVerb and resourcePath parts to wildcard', () => {
+    // ACT
+    const result = sut('arn:aws:execute-api:eu-west-1:920842203002:y2rp6iy583/dev/GET/journal');
+
+    // ASSERT
+    expect(result).toEqual('arn:aws:execute-api:eu-west-1:920842203002:y2rp6iy583/dev/*/*');
+  });
+
+  it('changes appends httpVerb and resourcePath wildcards if not present', () => {
+    // ACT
+    const result = sut('arn:aws:execute-api:eu-west-1:920842203002:y2rp6iy583/dev');
+
+    // ASSERT
+    expect(result).toEqual('arn:aws:execute-api:eu-west-1:920842203002:y2rp6iy583/dev/*/*');
+  });
+
+  it('does not perform validation', () => {
+    // ACT
+    const result = sut('abc/xyz');
+
+    // ASSERT
+    expect(result).toEqual('abc/xyz/*/*');
+  });
+
+  it(
+    'does not care about anything other than having at least 2 parts in the input methodArn',
+    () => {
+      // ACT
+      const result = sut(
+        'arn:aws:execute-api:{regionId}:{accountId}:{appId}/{stage}/{httpVerb}/{resrc}/{child}');
+
+      // ASSERT
+      expect(result)
+        .toEqual('arn:aws:execute-api:{regionId}:{accountId}:{appId}/{stage}/*/*');
+    });
+
+  it('throws error if stage name is not present', () => {
+    expect(() => sut('abc'))
+      .toThrow(new Error('Failed to extract arnBase and stageName from methodArn: "abc"'));
+  });
+
+  it('throws error if passed empty string', () => {
+    expect(() => sut(''))
+      .toThrow(new Error('Failed to extract arnBase and stageName from methodArn: ""'));
+  });
+
+  it('throws error if passed whitespace string', () => {
+    expect(() => sut(' \t '))
+      .toThrow(new Error('Failed to extract arnBase and stageName from methodArn: " \t "'));
+  });
+
+  it('throws error if passed null string', () => {
```

Comment:
Do we have strictness set high enough that the compiler wouldn't allow spurious nulls/undefined?

Reply:
@johnmorleybjss Its a good point, yes the strictness would pick up nulls passed in the Typescript.  The problem here is that the `methodArn` originates from outside of this service, the `methodArn` is passed by the AWS infrastructure.  Although I expect AWS or its configuration never to change to an extent that it cause a null to be passed, what I wanted here was a simple check to ensure we didn't get horrible to debug null reference errors.

Reply:
I see that the type of `CustomAuthorizerEvent.methodArn` is `string`, not `string | null | undefined`, so with the strictness levels as they are, there's no way a null/undefined can sneak through at compile time. However, banking on this at runtime seems like asking for trouble. Happy to leave it as is for safety's sake. 👍